server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -281,6 +289,52 @@ private RolloverResult rolloverDataStream(
             return new RolloverResult(newWriteIndexName, originalWriteIndex.getName(), currentState);
         }
 
+        DataStreamAutoShardingEvent dataStreamAutoShardingEvent = autoShardingResult != null ? switch (autoShardingResult.type()) {
```

Comment:
Super minor nit, but it's strange to have a trailing ` : dataStream.getAutoShardingEvent()` so far away from the ternary operator. Maybe we can switch it to be:

```java
DataStreamAutoShardingEvent dataStreamAutoShardingEvent = autoShardingResult == null ? dataStream.getAutoShardingEvent() : <big-long-switch-statement>
```

So it doesn't have the trailing ` : …` 30 lines away